### ISSUE TITLE
Warning: ...doesn’t export a component - False positive with ternary.

### ISSUE THREAD

**Issue by jsphstls** [styleguidist/react-styleguidist#884]
Version:
react-styleguidist@6.2.2

File:
https://gist.github.com/jsphstls/e2c8617f3c67cbc7b2a2854e389e1678
Location:
src\components\TestComponent\TestComponent.js

Warning: src\components\TestComponent\TestComponent.js matches a pattern defined in “components” or “sections” options in your style guide config but doesn’t export a component.

File contents:
```javascript
import PropTypes from 'prop-types'
import React from 'react'

const TestComponent = ({ bool }) => bool
  ? <p> True  </p>
  : <p> False </p>

TestComponent.propTypes = {
  bool: PropTypes.bool
}

export default TestComponent
```

If the ternary is not used, there is no warning. The component functions as expected regardless of the warning.

**Comment by sapegin** [comment]
I think this is react-docgen issue and should be reported there.

**Comment by jsphstls** [comment]
@sapegin I do not use react-docgen directly. How can I properly describe the issue in the context of react-docgen so it does not continue to be a defect of react-styleguidist?

**Comment by sapegin** [comment]
You can reproduce it like this:

```js
var reactDocs = require('react-docgen');
console.log(
	'1',
	reactDocs.parse(`
import PropTypes from 'prop-types'
import React from 'react'

const TestComponent = ({ bool }) => <p> True  </p>

TestComponent.propTypes = {
  bool: PropTypes.bool
}

export default TestComponent
`)
);
console.log(
	'2',
	reactDocs.parse(`
import PropTypes from 'prop-types'
import React from 'react'

const TestComponent = ({ bool }) => bool
  ? <p> True  </p>
  : <p> False </p>

TestComponent.propTypes = {
  bool: PropTypes.bool
}

export default TestComponent
`)
);

```
